app/src/test/java/nerd/tuxmobil/fahrplan/congress/schedule/FahrplanFragmentLoadLectureListTest.kt

```diff
@@ -0,0 +1,277 @@
+        whenever(lecturesDatabaseRepository.queryLecturesForDayIndexOrderedByDateUtc(dayIndex)) doReturn lectures
+
+        FahrplanFragment.loadLectureList(NoLogging, testableAppRepository, FahrplanFragment.rooms, dayIndex, false, onDoneDoNothing)
+
+        val expectedLectures = listOf(
+                Lecture("L0").apply {
+                    room = "Ada"
+                    roomIndex = 17
+                },
+                Lecture("L1").apply {
+                    room = "Ada"
+                    roomIndex = 17 // This room index is updated.
+                }
+        )
+        val expectedRoomList = SparseArrayCompat<Int>().apply { append(0, 17) }
+        assertThat(MyApp.lectureList as List<*>).isEqualTo(expectedLectures)
+        assertThat(MyApp.lectureListDay).isEqualTo(dayIndex)
+        assertThat(MyApp.roomsMap).isEqualTo(mapOf("Ada" to 17))
+        assertThat(MyApp.roomList.toString()).isEqualTo(expectedRoomList.toString())
+        assertThat(MyApp.roomCount).isEqualTo(1)
+    }
+
+    @Test
+    fun `Applies a new "roomIndex" to a lecture when its "roomIndex" is already taken`() {
+        MyApp.lectureList = null
+        MyApp.lectureListDay = 0
+        MyApp.roomCount = 0
+        MyApp.roomsMap["Borg"] = 42
+        MyApp.roomList.append(0, 42)
+        val dayIndex = 3
+        val lecture0 = DatabaseLecture(eventId = "L0", room = "Turing", roomIndex = 17)
+        val lecture1 = DatabaseLecture(eventId = "L1", room = "Lovelace", roomIndex = 17)
+        val lectures = listOf(lecture0, lecture1)
+        whenever(lecturesDatabaseRepository.queryLecturesForDayIndexOrderedByDateUtc(dayIndex)) doReturn lectures
+
+        FahrplanFragment.loadLectureList(NoLogging, testableAppRepository, FahrplanFragment.rooms, dayIndex, false, onDoneDoNothing)
+
+        val expectedLectures = listOf(
+                Lecture("L0").apply {
+                    room = "Turing"
+                    roomIndex = 17
+                },
+                Lecture("L1").apply {
+                    room = "Lovelace"
+                    roomIndex = 0 // This room index is updated.
+                }
+        )
+        val expectedRoomList = SparseArrayCompat<Int>().apply {
+            append(0, 0)
+            append(1, 17)
+        }
+        assertThat(lecture1.roomIndex).isEqualTo(17) // Stays unmodified because values are copied in List<LectureDatabaseModel>.toLecturesAppModel()
+        assertThat(MyApp.lectureList as List<*>).isEqualTo(expectedLectures)
+        assertThat(MyApp.lectureListDay).isEqualTo(dayIndex)
+        assertThat(MyApp.roomsMap).isEqualTo(mapOf("Turing" to 17, "Lovelace" to 0))
+        assertThat(MyApp.roomList.toString()).isEqualTo(expectedRoomList.toString())
+        assertThat(MyApp.roomCount).isEqualTo(2)
+    }
+
+    @Test
+    fun `Applies a new "roomIndex" to a lecture which takes place in a prioritized room and prepends the room in the "roomList"`() {
```

Comment:
It seems like the prioritizedRooms thing is broken. As far as I can tell the roomIndex is generated when parsing the schedule XML (it really should be a post-processing step, if at all). But that code doesn't consult `FahrplanFragment.rooms` at all. So it's possible for a non-prioritized room to get assigned a roomIndex value that should be reserved for prioritized rooms and for prioritized room to get a roomIndex that doesn't match the index in `FahrplanFragment.rooms`. So rooms aren't prioritized at all unless two rooms share the same roomIndex in the database (still not sure how this could happen or why there's code to handle this). And if that happens you could end up with two rooms sharing the same roomIndex again, e.g.

```kotlin
        val lecture0 = DatabaseLecture(eventId = "L0", room = "Ada", roomIndex = 2)
        val lecture1 = DatabaseLecture(eventId = "L1", room = "Hall C", roomIndex = 2)
```

will lead to both `Lecture` instances having a `roomIndex` value of 2 because the duplicate roomIndex value for "Hall C" is replaced with the index of "Hall C" in `FahrplanFragment.rooms`, which is also 2.

Reply:
Outdated due to https://github.com/EventFahrplan/EventFahrplan/pull/268 and the revised tests in 72bdbfd1bf56055762f0a1386d62b5592845914a.